packages/frontend-2/lib/viewer/composables/filtering/counts.ts

```diff
@@ -0,0 +1,189 @@
+import { FilteringExtension, type PropertyInfo, ViewerEvent } from '@speckle/viewer'
+import { useInjectedViewerState } from '~~/lib/viewer/composables/setup'
+import {
+  ExistenceFilterCondition,
+  type FilterData
+} from '~/lib/viewer/helpers/filters/types'
+
+/**
+ * Get count of filtered objects directly from the viewer
+ * Uses viewer events to stay in sync with the viewer's internal state
+ */
+export function useFilteredObjectsCount() {
+  const {
+    viewer,
+    ui: { filters }
+  } = useInjectedViewerState()
+  const filteredObjectsCount = ref(0)
+
+  const updateCount = () => {
+    const filteringExtension = viewer.instance.getExtension(FilteringExtension)
+    if (!filteringExtension) return
+
+    const isolatedObjects = filteringExtension.filteringState.isolatedObjects
+
+    const hasAppliedFilters = filters.propertyFilters.value.some(
+      (f) =>
+        f.isApplied &&
+        (f.selectedValues.length > 0 ||
+          ('isDefaultAllSelected' in f && f.isDefaultAllSelected))
+    )
+
+    if (!hasAppliedFilters) {
+      filteredObjectsCount.value = 0
+      return
+    }
+
+    const rawCount = isolatedObjects?.length || 0
+
+    // Ghost object that is used to represent objects that don't match the filter
+    const isGhostOnly = rawCount === 1 && isolatedObjects?.[0] === 'no-match-ghost-all'
+
+    if (isGhostOnly) {
+      filteredObjectsCount.value = 0
+      return
+    }
+
+    const realObjectCount =
+      isolatedObjects?.filter((id) => id !== 'no-match-ghost-all').length || 0
+    filteredObjectsCount.value = realObjectCount
+  }
+
+  onMounted(() => {
+    const filteringExtension = viewer.instance.getExtension(FilteringExtension)
+
+    filteringExtension.on(ViewerEvent.FilteringStateSet, updateCount)
+
+    updateCount()
+  })
+
+  onBeforeUnmount(() => {
+    const filteringExtension = viewer.instance?.getExtension(FilteringExtension)
+    if (filteringExtension) {
+      filteringExtension.removeListener(ViewerEvent.FilteringStateSet, updateCount)
+    }
+  })
+
+  return {
+    filteredObjectsCount: readonly(filteredObjectsCount)
+  }
+}
+
+let valueGroupCountCache = new WeakMap<PropertyInfo, Map<string, number>>()
```

Comment:
get rid of all module-level variables in all files. even if we don't currently load any of this in SSR, the moment this directly/indirectly gets evaluated during SSR, its gonna have the same variable reused across all requests and also potentially cause memory leaks

**always** use useState (if u want the state to transfer over from SSR) or useScopedState (if u dont need the value to transfer over from SSR)

https://vuejs.org/guide/scaling-up/ssr.html#cross-request-state-pollution